API.md

```diff
@@ -0,0 +1,445 @@
+# API
+
+## Table of Contents
+
+* [Core](#core)
+  - [Paginator](#paginator)
+    * [paginator.page](#paginatorpage)
+    * [paginator.queryHandler](#paginatorqueryhandler)
+    * [paginator.current](#paginatorcurrent)
+    * [paginator.next](#paginatornext)
+    * [paginator.prev](#paginatorprev)
+    * [paginator.first](#paginatorfirst)
+    * [paginator.last](#paginatorlast)
+    * [paginator.fetchPage](#paginatorfetchpage)
+    * [paginator.fetchPageCount](#paginatorfetchpagecount)
+    * [paginator.setQueryHandler](#paginatorsetqueryhandler)
+    * [paginator.setQueryParams](#paginatorsetqueryparams)
+  - [Query Handler](#query-handler)
+    * [queryHandler.makeParams](#queryhandlermakeparams)
+    * [queryHandler.resolvePage](#queryhandlerresolvepage)
+    * [queryHandler.setParams](#queryhandlersetparams)
+  - [PageMerger](#pagemerger)
+    * [new PageMerger](#new-pagemerger)
+    * [pageMerger.merge](#pagemergermerge)
+  - [PaginatorError](#paginatorerror)
+    * [new PaginatorError](#new-paginatorerror)
+* [Paginators](#paginators)
+  - [BasePaginator](#basepaginator)
+    * [new BasePaginator](#new-basepaginator)
+  - [PageNumberPaginator](#pagenumberpaginator)
+    * [new PageNumberPaginator](#new-pagenumberpaginator)
+  - [LimitOffsetPaginator](#limitoffsetpaginator)
+    * [new LimitOffsetPaginator](#new-limitoffsetpaginator)
+* [Query Handlers](#query-handlers)
+  - [PageNumberQueryHandler](#pagenumberqueryhandler)
+    * [new PageNumberQueryHandler](#new-pagenumberqueryhandler)
+    * [queryHandler.setOptions](#queryhandlersetoptions)
+  - [LimitOffsetQueryHandler](#limitoffsetqueryhandler)
+    * [new LimitOffsetQueryHandler](#new-limitoffsetqueryhandler)
+    * [queryHandler.setOptions](#queryhandlersetoptions-1)
+* [Utility](#utility)
+  - [paginate](#paginate)
+  - [all](#all)
+* [Requests](#requests)
+* [Comprehensive Example](#comprehensive-example)
+
+## Core
+
+### Paginator
+
+#### paginator.page
+
+This is a [getter][mdn-defineproperty] property.
+
+[mdn-defineproperty]: https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Object/defineProperty
+
+```javascript
+let paginator = new BasePaginator(request);
+// True
```

Comment:
You could say `-> True` or `returns: True` to make the meaning of the comment more obvious